test/letters/containers/AddressSection.unit.spec.jsx

```diff
@@ -141,4 +159,156 @@ describe('<AddressSection>', () => {
+    // Clear out country to get a validation error
+    tree.fillData('[name="country"]', '');
+    expect(tree.getElement('.usa-input-error')).to.not.be.null;
+  });
+
+  it('should infer new address type', () => {
+    const tree = SkinDeep.shallowRender(<AddressSection {...defaultProps}/>);
+
+    // Sanity check; make sure the type is what we expect before we change it
+    // NOTE: We're checking that it's domestic specifically just so we make absolutely sure
+    //  it's getting _changed_ to international instead of accidentally starting off as
+    //  international. Just a bit of future-proofing.
+    const instance = tree.getMountedInstance();
+    expect(instance.state.editableAddress.type).to.equal(ADDRESS_TYPES.domestic);
+
+    // Change the country so it'll be international
+    instance.handleChange('countryName', 'Elsweyre');
+    expect(instance.state.editableAddress.type).to.equal(ADDRESS_TYPES.international);
+
+    // NOTE: This isn't a _comprehensive_ test that ensures changing the input will do what we
+    //  expect, but the e2e test should make sure that the wiring from the input to handleChange
+    //  is up and running as expected.
+  });
+
+  it('should reset disallowed address fields when type changes', () => {
+    const tree = SkinDeep.shallowRender(<AddressSection {...defaultProps}/>);
+
+    // Sanity check; make sure the type is what we expect before we change it
+    const instance = tree.getMountedInstance();
+    expect(instance.state.editableAddress.stateCode).to.not.equal('');
+
+    // Change the country so it'll be international (and the state and zip fields should reset to '')
+    instance.handleChange('countryName', 'Elsweyre');
+    expect(instance.state.editableAddress.stateCode).to.equal('');
+  });
+
+  // Not sure how to test this bit yet...
+  // it('should scroll to first error', () => {});
+
+  describe('validation', () => {
+    // Spy on all the validation functions!
+    beforeEach(spyOnValidators);
+
+    // Clean up the spies so nobody finds out
+    afterEach(cleanUpSpies);
+
+    it('should run all validations against a field if the address is valid', () => {
+      const tree = SkinDeep.shallowRender(<AddressSection {...defaultProps}/>);
+      const instance = tree.getMountedInstance();
+
+      instance.handleChange('city', 'Elsweyre');
+      expect(AddressSection.fieldValidations.city.every(validator => validator.called)).to.be.true;
+    });
+
+    it('should return the first error message it finds', () => {
+      const tree = SkinDeep.shallowRender(<AddressSection {...defaultProps}/>);
+      const instance = tree.getMountedInstance();
+
+      instance.handleChange('city', '');
+      // The required validator (first in the list) should return an error message and no other validators should run
+      expect(AddressSection.fieldValidations.city[0].called).to.be.true;
+      expect(AddressSection.fieldValidations.city.every(validator => validator.called)).to.be.false;
+    });
```

Comment:
Since we're only expecting the first validator to be called, should this check just look to make sure that `AddressSection.fieldValidations.city[1]` wasn't called?

Alternatively, we could expect that every validator except the first was *not* called. That might be more comprehensive?